src/mlpack/core/optimizers/parallel_sgd/sgdp_impl.hpp

```diff
@@ -39,28 +39,38 @@ double ParallelSGD<DecomposableFunctionType>::Optimize(arma::mat& iterate)
   double lastObjective = DBL_MAX;
 
   //get maximum number of threads that will be running. with is defined by OMP_NUM_THREADS
-  int num_thread=omp_get_max_threads();
+  size_t num_thread=omp_get_max_threads();
+
+  //T: after T iteration it will go for tollerence checking;  we can define T in many way
+  //size_t T=maxIterations/num_thread;
+  //Here I have thaken T=contant
+  size_t T=1000;
```

Comment:
T can be a function of number of Thread , maxIterations, current  Iteration value . Here I have taken Constant.  If T=1. It will check tolerance each time which will slow down the algorithm.